modin/core/dataframe/algebra/binary.py

```diff
@@ -230,13 +245,25 @@ def caller(
                         elif infer_dtypes == "float":
                             dtypes = compute_dtypes_common_cast(query_compiler, other)
                             dtypes = dtypes.apply(coerce_int_to_float64)
+                    if (
+                        query_compiler._modin_frame._columns_cache is not None
+                        and other._modin_frame._columns_cache is not None
+                    ):
+                        if (
+                            len(query_compiler.columns) == 1
+                            and len(other.columns) == 1
+                            and len(query_compiler.columns.difference(other.columns))
+                            == 0
```

Comment:
would that work? using `.equals` looks simplier
```suggestion
                            and query_compiler.columns.equals(other.columns)
```

Reply:
fixed